packages/types/src/abort-handler.ts

```diff
@@ -0,0 +1,8 @@
+import type { AbortSignal as DeprecatedAbortSignal } from "./abort";
+
+/**
+ * @public
+ */
+export interface AbortHandler {
+  (this: AbortSignal | DeprecatedAbortSignal, ev: any): any;
+}
```

Comment:
nit question: will this ever need to be scoped down or have extra arguments?

Reply:
this should be `ev: Event -> void`, but I wouldn't change it at this point for compatibility